lib/src/tree.rs

```diff
@@ -534,11 +550,8 @@ pub fn merge_trees(
     assert_eq!(side1_tree.dir(), dir);
     assert_eq!(side2_tree.dir(), dir);
 
-    if base_tree.id() == side1_tree.id() {
-        return Ok(side2_tree.id().clone());
-    }
-    if base_tree.id() == side2_tree.id() || side1_tree.id() == side2_tree.id() {
-        return Ok(side1_tree.id().clone());
+    if let Some(resolved) = trivial_merge(&[side1_tree, base_tree, side2_tree]) {
+        return Ok(resolved.id.clone());
```

Comment:
I wasn't sure how future-proof it is to implement `PartialEq` and `Hash` based off only the id member, but it looks like we might not even need them? These lines suggest that we could just pass the tree ids to `trivial_merge()` instead of the full tree object.

Reply:
I think it is future-proof, and we do the same thing for `Commit` already, but good point that it's not even needed at this point. I've switched to passing just the ids.